staging/src/k8s.io/legacy-cloud-providers/azure/azure_standard.go

```diff
@@ -504,6 +504,20 @@ func (as *availabilitySet) GetPowerStatusByNodeName(name string) (powerState str
 	return vmPowerStateStopped, nil
 }
 
+// GetProvisioningStateByNodeName returns the provisioningState for the specified node.
+func (as *availabilitySet) GetProvisioningStateByNodeName(name string) (provisioningState string, err error) {
```

Comment:
Does this need to be public? It seems as though this should be private especially as a local bug fix.

Reply:
yes, because the function is part of `VMSet` interface.

Reply:
All other functions in the `VMSet` interface are public as well.

Reply:
The concern is related to the no feature development in legacy-cloud-providers. The idea is that we both want to encourage movement out of K/K for the in-tree providers and also we want to ensure the scope of the work for the extraction does not expand. Adding public methods suggest a problem with both of these. Interested in @andrewsykim thought on this?